src/Stryker.Core/Stryker.Core/Baseline/AzureFileShareBaselineProvider.cs

```diff
@@ -0,0 +1,212 @@
+﻿using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Stryker.Core.Logging;
+using Stryker.Core.Options;
+using Stryker.Core.Reporters.Json;
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Stryker.Core.Baseline
+{
+    public class AzureFileShareBaselineProvider : IBaselineProvider
+    {
+        private readonly StrykerOptions _options;
+        private readonly HttpClient _httpClient;
+        private readonly ILogger<AzureFileShareBaselineProvider> _logger;
+        public AzureFileShareBaselineProvider(StrykerOptions options, HttpClient httpClient = null)
+        {
+            _options = options;
+            _httpClient = httpClient ?? new HttpClient();
+            _logger = ApplicationLogging.LoggerFactory.CreateLogger<AzureFileShareBaselineProvider>();
+        }
+
+        public async Task<JsonReport> Load(string version)
+        {
+            var subdir = $"strykerOutput/baselines/{version}";
+            var filename = "stryker-report.json";
+            var url = new Uri($"{_options.AzureFileStorageUrl}/{subdir}/{filename}?sv={_options.AzureSAS}");
+
+            var requestMessage = new HttpRequestMessage(HttpMethod.Get, url);
+
+            requestMessage.Headers.Add("x-ms-date", "now");
+
+            var response = await _httpClient.SendAsync(requestMessage);
+
+            if (response.StatusCode == HttpStatusCode.OK)
+            {
+                var content = await response.Content.ReadAsStringAsync();
+
+                return JsonConvert.DeserializeObject<JsonReport>(content);
+            }
+            return null;
+        }
+
+        public async Task Save(JsonReport report, string version)
+        {
+            var existingReport = await Load(version);
+
+            var reportJson = report.ToJson();
+
+            int byteSize = Encoding.UTF8.GetByteCount(report.ToJson());
+
+            if (existingReport == null)
+            {
+                var succesfullyCreatedDirectory = await CreateDirectory(version);
+
+                if (!succesfullyCreatedDirectory)
+                {
+                    return;
+                }
+
+                var successfullyAllocated = await AllocateFileLocation(byteSize, version);
+
+                if (!successfullyAllocated)
+                {
+                    return;
+                }
+            }
+            else if (Encoding.UTF8.GetByteCount(existingReport.ToJson()) != byteSize)
+            {
+                var succesfullyResizedFile = await ResizeFile(byteSize, version);
+
+                if (!succesfullyResizedFile)
+                {
+                    return;
+                }
+            }
+
+
+            await UploadFile(reportJson, byteSize, version);
+        }
+
+        public async Task<bool> CreateDirectory(string version)
+        {
+            var subdir = $"strykerOutput/baselines/{version}";
```

Comment:
When trying out the feature I get a notfound. When trying to create the directories strykerOutput/baselines in the portal it says I can't have a / in the name so it looks like you have to create the sub-directories one-by-one.